Updates according to the JMTE Y3 report

whyjz opened the pull request:
Also re-organized the Jupyter Book layout; add relevant external resources in the TOC.

@fperez what do you think about making the slides used in the 2022 EC workshop available to the public?

consideRatio commented:
Using git blame I found this:

```
# sphinxcontrib-bibtex pinning reason: https://github.com/executablebooks/jupyter-book/issues/1137
sphinxcontrib-bibtex<2.0.0
```

I think maybe the use of this extension may need explicit configuration of where the .bib file is located. See https://github.com/executablebooks/jupyter-book/pull/1754/files.

I'm not sure why there are build errors, but since I see a very ancient pinning of a Python package, I assume that can be related. Let's try get that unpinned!

consideRatio commented:
@whyjz I added the commit [deae4d2](https://github.com/pangeo-data/jupyter-earth/pull/149/commits/deae4d27b5deaca21e72b26b9a22a8e03d59cd66) to this PR, which resolved the build failure!

whyjz commented:
Thank you @consideRatio ! I was about the take a look today but realized you have beautifully solved the dependency issue. We are the best team!

whyjz commented:
I will leave this PR open for one more day in case anyone wants to review it, then I will merge and publish the updates.

consideRatio commented on .github/workflows/build-doc.yml:
```diff
@@ -21,12 +21,12 @@ jobs:
         update-conda: true
         conda-channels: conda-forge
-        python-version: 3.7
+        python-version: 3.8
```
By the time we enter `python-version: 3.10`, this YAML will parse to the number `3.1` and then we are in trouble :)

```suggestion
        python-version: "3.8"
```

consideRatio commented on docs/_toc.yml:
```diff
@@ -1,34 +1,54 @@
+        - file: jupyter-resources/introduction/overview
+        - file: jupyter-resources/introduction/ecosystem
+        - file: jupyter-resources/introduction/tour
+    # - file: jupyter-resources/quickstart/index
+      # sections:
+        # - file: jupyter-resources/quickstart/environment
+        # - file: jupyter-resources/quickstart/get-data
+        # - file: jupyter-resources/quickstart/interactive
+        # - file: jupyter-resources/quickstart/computing
+        # - file: jupyter-resources/quickstart/visualization
+        # - file: jupyter-resources/quickstart/sharing-science
+        # - file: jupyter-resources/quickstart/writing-doc
```
In case they should be deleted, here is a code suggestion about that.

```suggestion
```

consideRatio reviewed (commented):
Thank you for your work on this @whyjz!!!

whyjz commented:
Thanks Erik! I am running out of time today but will take a look tomorrow.

whyjz commented:
@consideRatio I have done addressing your comments!
For the quickstart section, yes, it is not finished yet, and we did not mention that in the Y3 report. So I just created another branch and moved the unfinished content there for another PR in the future.
